infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationClientChannelAdapter.java

```diff
@@ -380,6 +383,46 @@ public void onCompleted() {
         replicationReqObserverMap.get(Pair.of(sessionMsg, requestId)).onNext(request);
     }
 
+    private void remoteSessionRegister(String nodeId, RequestMsg request) {
```

Comment:
While its okay that it is using async call (because our LR logic is synchronous), the logging and the rpc should change to use the `remoteSessionRegister`.

We can also simplify this code to be like the negotiation request. But since this is a sample code, I'll leave it to you.